src/software/thunderscope/binary_context_managers/game_controller.py

```diff
@@ -21,29 +22,61 @@
 
 
 class Gamecontroller(object):
-    """ Gamecontroller Context Manager """
+    """Gamecontroller Context Manager"""
 
     CI_MODE_LAUNCH_DELAY_S = 0.3
     REFEREE_IP = "224.5.23.1"
     CI_MODE_OUTPUT_RECEIVE_BUFFER_SIZE = 9000
 
-    def __init__(self, supress_logs: bool = False) -> None:
+    def __init__(self, supress_logs: bool = False, use_conventional_port=False) -> None:
         """Run Gamecontroller
 
         :param supress_logs: Whether to suppress the logs
+        :not_launch_gc Whether to launch the gamecontroller or not!
         """
+
         self.supress_logs = supress_logs
 
-        # We need to find 2 free ports to use for the gamecontroller
-        # so that we can run multiple gamecontroller instances in parallel
-        self.referee_port = self.next_free_port()
         self.ci_port = self.next_free_port()
+        # we are not using conventional by default since most of the time
+        # we are not in competition and. Thus, we would be conflicting with
+        # each other if we are using conventional port
+        self.referee_port = self.next_free_port(random.randint(1024, 65535))
+        if use_conventional_port:
+            if not self.is_valid_port(40000):
+                raise OSError("Cannot use port 40000 for Gamecontroller")
+
+            self.referee_port = 40000
 
+        self.ci_port = self.next_free_port()
         # this allows gamecontroller to listen to override commands
         self.command_override_buffer = ThreadSafeBuffer(
             buffer_size=2, protobuf_type=ManualGCCommand
         )
 
+    @staticmethod
+    def get_referee_port_staticmethod(gamecontroller: Gamecontroller):
```

Comment:
```suggestion
    def get_referee_port(gamecontroller: Gamecontroller):
```
it doesn't need to be called `staticmethod`

Reply:
we kind of need a static method of a function. I feel like a static method would make more sense. see the comments below for more justification.